src/NuGet.Clients/NuGet.VisualStudio/Extensibility/IVsNuGetPathContext.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+
+namespace NuGet.VisualStudio
+{
+    /// <summary>
+    /// NuGet path information specific to the current context (e.g. project context).
+    /// </summary>
+    [ComImport]
+    [Guid("24A1A187-75EE-4296-A8B3-59F0E0707119")]
+    public interface IVsNuGetPathContext
+    {
+        /// <summary>
+        /// User package folder directory. The path returned is an absolute path.
+        /// </summary>
+        string UserPackageFolder { get; }
+
+        /// <summary>
+        /// Fallback package folder locations. The paths (if any) in the returned list are absolute paths. If no
+        /// fallback package folders are configured, an empty list is returned.
+        /// </summary>
+        IReadOnlyList<string> FallbackPackageFolders { get; }
```

Comment:
Neither those _legacy_ types are. The question is whether we want to start doing the right thing for new types. As we did for the `IVsSolutionRestoreService`. It obviously would require more work to do. As an alternative I'd suggest removing `[ComImport]` and `[Guid]` to avoid confusion at least.